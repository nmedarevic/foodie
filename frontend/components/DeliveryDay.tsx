import { Section } from "./Common/Section";

type DevliveryDayProps = {
  availableDays: string[];
  onClick: Function;
  selectedIndex: number;
}
export const DeliveryDay = ({
  availableDays = [],
  onClick = (index) => {},
  selectedIndex = 0
}: DevliveryDayProps) => (
  <Section>
    <div className='flex flex-col'>
      <h1 className='text-2xl'>Select first delivery day</h1>
      <div className='flex'>
        {
          availableDays.map((day, index) => (
            <div
              data-test={day}
              onClick={() => onClick(index)}
              key={day}
              className={`mt-5 mr-5 cursor-pointer nline-flex items-center px-6 py-3 ${selectedIndex === index ? 'text-white' : 'text-black'} font-semibold shadow-sm ${selectedIndex === index ? 'bg-blue-500' : 'bg-gray-200'}`}
            >
              {day}
            </div>
          ))
        }
      </div>
    </div>
  </Section>
)
