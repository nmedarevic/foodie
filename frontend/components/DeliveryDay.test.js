import React from 'react'
import renderer from 'react-test-renderer'
import {DeliveryDay} from './DeliveryDay'
const availableDays = ['one', 'two']

it('Renders Delivery day without props', () => {
  const tree = renderer.create(<DeliveryDay />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Renders Delivery day with basic props', () => {
  const tree = renderer.create(<DeliveryDay availableDays={availableDays}/>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Renders Delivery day with a selected day', () => {
  const tree = renderer.create(<DeliveryDay availableDays={availableDays} selectedIndex={1}/>).toJSON()
  expect(tree).toMatchSnapshot()
})
