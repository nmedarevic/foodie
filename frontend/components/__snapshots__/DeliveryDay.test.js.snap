// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Delivery day with a selected day 1`] = `
<section
  className="bg-white py-8 flex"
>
  <div
    className="container mx-auto flex items-center flex-wrap flex-col pt-4 pb-12 self-center"
  >
    <div
      className="flex flex-col"
    >
      <h1
        className="text-2xl"
      >
        Select first delivery day
      </h1>
      <div
        className="flex"
      >
        <div
          className="mt-5 mr-5 cursor-pointer nline-flex items-center px-6 py-3 text-black font-semibold shadow-sm bg-gray-200"
          data-test="one"
          onClick={[Function]}
        >
          one
        </div>
        <div
          className="mt-5 mr-5 cursor-pointer nline-flex items-center px-6 py-3 text-white font-semibold shadow-sm bg-blue-500"
          data-test="two"
          onClick={[Function]}
        >
          two
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Renders Delivery day with basic props 1`] = `
<section
  className="bg-white py-8 flex"
>
  <div
    className="container mx-auto flex items-center flex-wrap flex-col pt-4 pb-12 self-center"
  >
    <div
      className="flex flex-col"
    >
      <h1
        className="text-2xl"
      >
        Select first delivery day
      </h1>
      <div
        className="flex"
      >
        <div
          className="mt-5 mr-5 cursor-pointer nline-flex items-center px-6 py-3 text-white font-semibold shadow-sm bg-blue-500"
          data-test="one"
          onClick={[Function]}
        >
          one
        </div>
        <div
          className="mt-5 mr-5 cursor-pointer nline-flex items-center px-6 py-3 text-black font-semibold shadow-sm bg-gray-200"
          data-test="two"
          onClick={[Function]}
        >
          two
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Renders Delivery day without props 1`] = `
<section
  className="bg-white py-8 flex"
>
  <div
    className="container mx-auto flex items-center flex-wrap flex-col pt-4 pb-12 self-center"
  >
    <div
      className="flex flex-col"
    >
      <h1
        className="text-2xl"
      >
        Select first delivery day
      </h1>
      <div
        className="flex"
      />
    </div>
  </div>
</section>
`;
